lib/liquid/parse_context.rb

```diff
@@ -0,0 +1,39 @@
+module Liquid
+  class ParseContext
+    attr_accessor :partial, :locale, :line_number
+    attr_reader :warnings, :error_mode
+
+    def initialize(options = {})
+      @template_options = options ? options.dup : {}
```

Comment:
Do you also want to set `@options` here so that it always exists even if `partial=` hasn't been called?